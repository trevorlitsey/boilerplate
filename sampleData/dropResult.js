const dropResult = {
	draggableId: 'jgux6edn',
	type: 'DEFAULT',
	source: {
		droppableId: 'draftOrder',
		index: 0,
	},
	destination: {
		droppableId: 'draftOrder',
		index: 1,
	},
	reason: 'DROP',
}