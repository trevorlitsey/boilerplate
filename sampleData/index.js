export { default as preview } from './preview';
export { default as snippets } from './snippets';
export { default as dropResult } from './dropResult';