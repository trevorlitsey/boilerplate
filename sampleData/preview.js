const preview = {
	draftOrder: ['jgux6edn', 'jghgygc9'],
	snippetOrder: ['jguxf292'],
}

export default preview;