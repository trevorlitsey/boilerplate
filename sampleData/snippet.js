const snippet = {
	title: 'Section Two',
	text: 'Praesent aliquet placerat augue in varius. Sed lorem metus, gravida at lectus nec, egestas convallis ex. Suspendisse accumsan quam nec laoreet fermentum. Donec scelerisque in lorem non aliquet. Nunc vestibulum eleifend felis eget dignissim. Integer ullamcorper ipsum ac imperdiet varius. Maecenas cursus egestas neque, quis placerat enim dignissim eu. Integer ultrices aliquam metus eget faucibus. Vivamus lobortis turpis odio, sed sodales odio elementum nec. Mauris facilisis ipsum et ante porta accumsan. Quisque molestie maximus leo, id cursus odio porttitor mattis. Quisque eu porttitor justo. Ut facilisis pretium arcu, eget mollis sem mattis ut.',
	tags: [
		{ id: 'jgtv2988', label: 'lots', value: 'lots' },
		{ id: 'jgtv29wa', label: 'of', value: 'of' },
		{ id: 'jgtv1e21', label: 'tags', value: 'tags' },
	]
}

export default snippet;