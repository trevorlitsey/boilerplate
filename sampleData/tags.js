const tags = [
	{ id: 'jgtv2988', label: 'lots', value: 'lots' },
	{ id: 'jgtv29wa', label: 'of', value: 'of' },
	{ id: 'jgtv1e21', label: 'tags', value: 'tags' },
]

export default tags;