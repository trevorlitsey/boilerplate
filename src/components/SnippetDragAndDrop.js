import React from 'react';
import styled from 'styled-components';
import ReactTooltip from 'react-tooltip'
import { Draggable, Droppable } from 'react-beautiful-dnd';

const Container = styled.div`
	display: flex;
	flex-wrap: wrap;
	background: var(--light);
	padding: 10px;
	margin-bottom: 20px;

	& > .card {
		padding: 8px;
		margin: 2px;
	}

	.icon:hover {
		cursor: pointer;
	}

`

const grid = 8;

const getItemStyle = (isDragging, draggableStyle) => ({
	// some basic styles to make the items look a bit nicer
	userSelect: 'none',
	padding: grid * 2,
	margin: `0 ${grid}px 0 0`,
	lineHeight: 1,
	minWidth: 100,
	textAlign: 'center',
	display: 'inline-block',

	// styles we need to apply on draggables
	...draggableStyle,
});

const getListStyle = isDraggingOver => ({
	background: isDraggingOver ? 'var(--gray)' : 'var(--light)',
	padding: grid,
	transition: 'all .3s',
	display: 'flex',
	overflow: 'scroll',
});

class SnippetDragAndDrop extends React.PureComponent {

	render() {

		const { snippets } = this.props;

		return (
			<Droppable droppableId="snippets" direction="horizontal">
				{(provided, snapshot) => (
					<div
						className="rounded"
						ref={provided.innerRef}
						style={getListStyle(snapshot.isDraggingOver)}
					>
						{snippets.map((snippet, index) => (
							<Draggable key={snippet.id} draggableId={snippet.id} index={index}>
								{(provided, snapshot) => (
									<div>
										<div
											className="card"
											data-tip={snippet.content.length > 30 ? snippet.content.substring(0, 30) + ' ...' : snippet.content}
											ref={provided.innerRef}
											{...provided.draggableProps}
											{...provided.dragHandleProps}
											style={getItemStyle(
												snapshot.isDragging,
												provided.draggableProps.style
											)}
										>
											<p
												className="card-text"
											>
												{snippet.title}
											</p>
										</div>
										<ReactTooltip effect="solid" />
									</div>
								)}
							</Draggable>
						))}
						{provided.placeholder}
					</div>
				)}
			</Droppable>
		)
	}
}

export default SnippetDragAndDrop;