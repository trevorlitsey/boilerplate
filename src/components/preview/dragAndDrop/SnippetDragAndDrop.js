import React from 'react';
import { object, array } from 'prop-types';
import styled from 'styled-components';
import ReactTooltip from 'react-tooltip'
import { Draggable, Droppable } from 'react-beautiful-dnd';
import { Link } from 'react-router-dom'

import Spinner from '../../shared/Spinner';

import { GrayH4 } from '../../../styles/components';

import { getSnippetItemStyle, getSnippetListStyle } from './styles';

const ScrollContainer = styled.div`
	width: 100%;
	overflow: scroll;
`

class SnippetDragAndDrop extends React.PureComponent {

	static propTypes = {
		snippets: object.isRequired,
		snippetOrder: array.isRequired,
	}

	render() {

		const { snippets, snippetOrder, loading } = this.props;

		if (loading) {
			return (
				<Droppable droppableId="snippetOrder" direction="horizontal">
					{(provided, snapshot) => (
						<div
							className="rounded"
							ref={provided.innerRef}
							style={getSnippetListStyle(snapshot.isDraggingOver)}
						>
							<Spinner style={{ margin: 'auto' }} />
						</div>
					)}
				</Droppable>
			)
		}

		if (!snippetOrder.length) {
			return (
				<Droppable droppableId="snippetOrder" direction="horizontal">
					{(provided, snapshot) => (
						<div
							className="rounded"
							ref={provided.innerRef}
							style={getSnippetListStyle(snapshot.isDraggingOver)}
						>
							<h5 style={{ margin: 'auto', color: 'var(--gray)' }}>Click <Link to="/snippets">here</Link> to add snippets.</h5>
						</div>
					)}
				</Droppable>
			)
		}

		return (
			<ScrollContainer>
				<Droppable droppableId="snippetOrder" direction="horizontal">
					{(provided, snapshot) => (
						<div
							className="rounded"
							data-test="snippetOrder"
							ref={provided.innerRef}
							style={getSnippetListStyle(snapshot.isDraggingOver)}
						>
							{snippetOrder && snippetOrder.map((id, index) => {

								const snippet = snippets[id];

								const { title, text } = snippet;

								return (
									<Draggable key={id} draggableId={id} index={index}>
										{(provided, snapshot) => (
											<div>
												<div
													className="card"
													data-tip={text.length > 30 ? text.substring(0, 30) + ' ...' : text}
													ref={provided.innerRef}
													{...provided.draggableProps}
													{...provided.dragHandleProps}
													style={getSnippetItemStyle(
														snapshot.isDragging,
														provided.draggableProps.style
													)}
												>
													<p className="card-text">{title}</p>
												</div>
												<ReactTooltip effect="solid" />
											</div>
										)}
									</Draggable>
								)
							})}
							{provided.placeholder}
						</div>
					)}
				</Droppable>
			</ScrollContainer>
		)
	}
}

export default SnippetDragAndDrop;