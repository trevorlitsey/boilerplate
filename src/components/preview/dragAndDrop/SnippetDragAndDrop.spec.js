import React from 'react';
import { shallow } from 'enzyme';
import wait from 'waait';

import { snippets, preview } from '../../../../sampleData';

import SnippetDragAndDrop, { ScrollContainer } from './PreviewDragAndDrop'

describe('SnippetDragAndDrop', () => {

	it('should display spinner if loading', () => {
		const wrapper = renderSnippetDragAndDrop({ loading: true });
		expect(wrapper.find('[data-test="loading"]').length).toBe(1);
		expect(wrapper).toMatchSnapshot()
	})

	it('should should say CLICK HERE TO ADD SNIPPETS if no order is given', () => {
		const wrapper = renderSnippetDragAndDrop({ snippetOrder: [] })
		expect(wrapper).toMatchSnapshot()
		expect(wrapper.find('[data-test="no-order"]').length).toBe(1);
	})

	// TODO: report bug for this test
	xit('should render snippet cards when order is given', async () => {
		const wrapper = renderSnippetDragAndDrop();
		expect(wrapper.find('[data-test="render-order"]').length).toBe(1);
		expect(wrapper).toMatchSnapshot()
	})

})

// -------
function renderSnippetDragAndDrop(props = {}) {
	const propsToUser = {
		snippets,
		snippetOrder: preview.snippetOrder,
		loading: false,
		...props,
	}

	return shallow(<SnippetDragAndDrop {...propsToUser} />)
}