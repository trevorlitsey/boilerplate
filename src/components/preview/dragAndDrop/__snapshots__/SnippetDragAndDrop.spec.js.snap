// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnippetDragAndDrop should display spinner if loading 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SnippetDragAndDrop
    loading={true}
    snippetOrder={
      Array [
        "jguxf292",
      ]
    }
    snippets={
      Object {
        "jghgygc9": Object {
          "tags": Array [
            Object {
              "id": "jgtv2988",
              "label": "lots",
              "value": "lots",
            },
            Object {
              "id": "jgtv29wa",
              "label": "of",
              "value": "of",
            },
            Object {
              "id": "jgtv1e21",
              "label": "tags",
              "value": "tags",
            },
          ],
          "text": "Praesent aliquet placerat augue in varius. Sed lorem metus, gravida at lectus nec, egestas convallis ex. Suspendisse accumsan quam nec laoreet fermentum. Donec scelerisque in lorem non aliquet. Nunc vestibulum eleifend felis eget dignissim. Integer ullamcorper ipsum ac imperdiet varius. Maecenas cursus egestas neque, quis placerat enim dignissim eu. Integer ultrices aliquam metus eget faucibus. Vivamus lobortis turpis odio, sed sodales odio elementum nec. Mauris facilisis ipsum et ante porta accumsan. Quisque molestie maximus leo, id cursus odio porttitor mattis. Quisque eu porttitor justo. Ut facilisis pretium arcu, eget mollis sem mattis ut.",
          "title": "Section Two",
        },
        "jgux6edn": Object {
          "tags": Array [],
          "text": "e",
          "title": "e",
        },
        "jguxf292": Object {
          "tags": Array [],
          "text": "e",
          "title": "e!!!!????",
        },
        "jgxvm7zr": Object {
          "tags": Array [],
          "text": "yes",
          "title": "yes!",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
      "data-test": "loading",
      "direction": "horizontal",
      "droppableId": "snippetOrder",
      "ignoreContainerClipping": false,
      "isDropDisabled": false,
      "type": "DEFAULT",
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
        "data-test": "loading",
        "direction": "horizontal",
        "droppableId": "snippetOrder",
        "ignoreContainerClipping": false,
        "isDropDisabled": false,
        "type": "DEFAULT",
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SnippetDragAndDrop should render snippet cards when order is given 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SnippetDragAndDrop
    loading={false}
    snippetOrder={
      Array [
        "jguxf292",
      ]
    }
    snippets={
      Object {
        "jghgygc9": Object {
          "tags": Array [
            Object {
              "id": "jgtv2988",
              "label": "lots",
              "value": "lots",
            },
            Object {
              "id": "jgtv29wa",
              "label": "of",
              "value": "of",
            },
            Object {
              "id": "jgtv1e21",
              "label": "tags",
              "value": "tags",
            },
          ],
          "text": "Praesent aliquet placerat augue in varius. Sed lorem metus, gravida at lectus nec, egestas convallis ex. Suspendisse accumsan quam nec laoreet fermentum. Donec scelerisque in lorem non aliquet. Nunc vestibulum eleifend felis eget dignissim. Integer ullamcorper ipsum ac imperdiet varius. Maecenas cursus egestas neque, quis placerat enim dignissim eu. Integer ultrices aliquam metus eget faucibus. Vivamus lobortis turpis odio, sed sodales odio elementum nec. Mauris facilisis ipsum et ante porta accumsan. Quisque molestie maximus leo, id cursus odio porttitor mattis. Quisque eu porttitor justo. Ut facilisis pretium arcu, eget mollis sem mattis ut.",
          "title": "Section Two",
        },
        "jgux6edn": Object {
          "tags": Array [],
          "text": "e",
          "title": "e",
        },
        "jguxf292": Object {
          "tags": Array [],
          "text": "e",
          "title": "e!!!!????",
        },
        "jgxvm7zr": Object {
          "tags": Array [],
          "text": "yes",
          "title": "yes!",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Connect(Droppable)
        direction="horizontal"
        droppableId="snippetOrder"
        ignoreContainerClipping={false}
        isDropDisabled={false}
        type="DEFAULT"
      >
        [Function]
      </Connect(Droppable)>,
      "data-test": "render-order",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
        "direction": "horizontal",
        "droppableId": "snippetOrder",
        "ignoreContainerClipping": false,
        "isDropDisabled": false,
        "type": "DEFAULT",
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Connect(Droppable)
          direction="horizontal"
          droppableId="snippetOrder"
          ignoreContainerClipping={false}
          isDropDisabled={false}
          type="DEFAULT"
        >
          [Function]
        </Connect(Droppable)>,
        "data-test": "render-order",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": [Function],
          "direction": "horizontal",
          "droppableId": "snippetOrder",
          "ignoreContainerClipping": false,
          "isDropDisabled": false,
          "type": "DEFAULT",
        },
        "ref": null,
        "rendered": [Function],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SnippetDragAndDrop should should say CLICK HERE TO ADD SNIPPETS if no order is given 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SnippetDragAndDrop
    loading={false}
    snippetOrder={Array []}
    snippets={
      Object {
        "jghgygc9": Object {
          "tags": Array [
            Object {
              "id": "jgtv2988",
              "label": "lots",
              "value": "lots",
            },
            Object {
              "id": "jgtv29wa",
              "label": "of",
              "value": "of",
            },
            Object {
              "id": "jgtv1e21",
              "label": "tags",
              "value": "tags",
            },
          ],
          "text": "Praesent aliquet placerat augue in varius. Sed lorem metus, gravida at lectus nec, egestas convallis ex. Suspendisse accumsan quam nec laoreet fermentum. Donec scelerisque in lorem non aliquet. Nunc vestibulum eleifend felis eget dignissim. Integer ullamcorper ipsum ac imperdiet varius. Maecenas cursus egestas neque, quis placerat enim dignissim eu. Integer ultrices aliquam metus eget faucibus. Vivamus lobortis turpis odio, sed sodales odio elementum nec. Mauris facilisis ipsum et ante porta accumsan. Quisque molestie maximus leo, id cursus odio porttitor mattis. Quisque eu porttitor justo. Ut facilisis pretium arcu, eget mollis sem mattis ut.",
          "title": "Section Two",
        },
        "jgux6edn": Object {
          "tags": Array [],
          "text": "e",
          "title": "e",
        },
        "jguxf292": Object {
          "tags": Array [],
          "text": "e",
          "title": "e!!!!????",
        },
        "jgxvm7zr": Object {
          "tags": Array [],
          "text": "yes",
          "title": "yes!",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
      "data-test": "no-order",
      "direction": "horizontal",
      "droppableId": "snippetOrder",
      "ignoreContainerClipping": false,
      "isDropDisabled": false,
      "type": "DEFAULT",
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
        "data-test": "no-order",
        "direction": "horizontal",
        "droppableId": "snippetOrder",
        "ignoreContainerClipping": false,
        "isDropDisabled": false,
        "type": "DEFAULT",
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
